### ISSUE TITLE
Reverse proxy tries to bind to port 80 even when it does not need to

### ISSUE THREAD

**Issue by josvazg** [caddyserver/caddy#5510]
We doing a local reverse proxy with caddy so we can easily can test some UI doing OAuth login locally.

Our OAuth IDP does not like localhost domains, so for tests we trick the browser with an entry in the local `/etc/hosts` for an invented host (eg `mydevmachine.example.com`) in the same domain as the IDP, so that CORS is happy about it, and we also get a proper callback URL to an HTTPS  non-localhost site that "looks legit", from the IDP point of view.

We would like our tests _NOT to require_ **sudo** privileges and we were testing launching caddy with something like this:

```shell
caddy reverse-proxy --internal-certs --from sealed-secrets-ui.dev.vmware.com:8443 --to localhost:4200
```

But we get the following error:
```shell
Error: loading new config: http app module: start: listening on :80: listen tcp :80: bind: permission denied
```
Why is it trying to bind to port 80? we did not specify it at all, neither in the origin nor the target.

BTW, if we do that with sudo it works and the reverse proxy is correct... but if we check netstat we see it is listening to port 80 still:
```shell
$ sudo netstat -lepunt |grep caddy
tcp6       0      0 :::8443                 :::*                    LISTEN      0          2232068    475262/caddy        
tcp6       0      0 :::80                   :::*                    LISTEN      0          2232070    475262/caddy        
udp6       0      0 :::8443                 :::*                                0          2232069    475262/caddy  
```
Why?

**Comment by josvazg** [comment]
Ok I got it!

It works if you use `--disable-redirects`
I am guessing HTTP -> HTTPS redirects default to port `:80`.

Feel free to close the issue, unless you want to provide or ask more info. We are now unblocked.

**Comment by IndeedNotJames** [comment]
Yep, that's due to HTTP -> HTTPS redirects (and solving ACME HTTP challenges).

See https://caddyserver.com/docs/automatic-https#effects

Glad you figured it out yourself :)

**Comment by mholt** [comment]
Can confirm :) Thanks @IndeedNotJames

@josvazg Is your company using Caddy? We would love to best support your use case. Can I email you or your team regarding support options to see how we can help (or help in the future when you need it)?
